pkg/oracle/neofs/neofs.go

```diff
@@ -0,0 +1,136 @@
+package neofs
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/mr-tron/base58"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neofs-api-go/pkg/client"
+	"github.com/nspcc-dev/neofs-api-go/pkg/container"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+)
+
+const (
+	// URIScheme is the name of neofs URI scheme.
+	URIScheme = "neofs"
+
+	// containerIDSize is the size of container id in bytes.
+	containerIDSize = sha256.Size
+	// objectIDSize is the size of container id in bytes.
+	objectIDSize = sha256.Size
+	// rangeSep is a separator between offset and length.
+	rangeSep = '|'
+
+	rangeCmd  = "range"
+	headerCmd = "header"
+)
+
+// Various validation errors.
+var (
+	ErrInvalidScheme    = errors.New("invalid URI scheme")
+	ErrMissingObject    = errors.New("object ID is missing from URI")
+	ErrInvalidContainer = errors.New("container ID is invalid")
+	ErrInvalidObject    = errors.New("object ID is invalid")
+	ErrInvalidRange     = errors.New("object range is invalid (expected 'Offset|Length'")
+	ErrInvalidCommand   = errors.New("invalid command")
+)
+
+// Get returns neofs object from the provided url.
+// URI scheme is "neofs://<Container-ID>/<Object-ID/<Command>/<Params>".
+// If Command is not provided, full object is requested.
+// TODO multiple addresses
+// 	Addresses are queried sequentially using the same context.
+func Get(ctx context.Context, priv *keys.PrivateKey, u *url.URL, addr string) ([]byte, error) {
+	if u.Scheme != URIScheme {
+		return nil, ErrInvalidScheme
+	}
+	ps := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
+	if len(ps) == 0 {
+		return nil, ErrMissingObject
+	}
+	spew.Dump(ps)
```

Comment:
No spewing here, please.

Reply:
I will remove this after finish testing.

Reply:
Fixed.

Reply:
It's still there in tests and it adds an additional dependency.